src/materials/ConcreteASREigenstrain.C

```diff
@@ -0,0 +1,323 @@
+  params.addRangeCheckedParam<Real>("latency_activation_energy",
+                                    9400.0,
+                                    "latency_activation_energy > 0.0",
+                                    "Activation energy associated with latency_time (U_L)");
+  params.addRequiredParam<Real>("reference_temperature",
+                                "Reference temperature for ASR reaction constants.");
+
+  // Note that Fahrenheit is not supported because that would require different parameters for the
+  // times and activation energies
+  MooseEnum temperature_units("Celsius Kelvin");
+  params.addRequiredParam<MooseEnum>(
+      "temperature_unit",
+      temperature_units,
+      "Unit used to define 'temperature' and 'reference_temperature'");
+
+  params.addRangeCheckedParam<Real>(
+      "stress_latency_factor",
+      1.3333,
+      "stress_latency_factor >= 0.0",
+      "Constant for ASR latency time retardation under hydrostatic compression (alpha)");
+
+  params.addRangeCheckedParam<Real>(
+      "tensile_absorption_factor",
+      0.5,
+      "tensile_absorption_factor >= 0.0 & tensile_absorption_factor <= 1.0",
+      "Fraction of tensile strength beyond which ASR gel is absorbed into tensile cracks "
+      "(gamma_t)");
+  params.addRangeCheckedParam<Real>(
+      "tensile_retention_factor",
+      0.5,
+      "tensile_retention_factor >= 0.0 & tensile_retention_factor <= 1.0",
+      "Residual ASR retention factor under tension (gamma_r)");
+
+  params.addParam<Real>("compressive_stress_exponent",
+                        0.5,
+                        "Exponent for ASR retention factor under compressive stress state (beta)");
+
+  params.addParam<bool>("ASR_dependent_tensile_strength",
+                        false,
+                        "Set true to turn ASR reaction dependent tensile strength");
+  params.addRequiredRangeCheckedParam<Real>(
+      "residual_tensile_strength_fraction",
+      "residual_tensile_strength_fraction >= 0.0 & residual_tensile_strength_fraction <= 1.0",
+      "Residual fraction of tensile strength at full ASR reaction");
+
+  params.addParamNamesToGroup("stress_latency_factor tensile_absorption_factor "
+                              "tensile_retention_factor compressive_stress_exponent",
+                              "Advanced");
+
+  params.addParam<unsigned int>(
+      "max_its", 30, "Maximum number of iterations for material solution");
+  params.addParam<bool>(
+      "output_iteration_info", false, "Set true to output material iteration information");
+  params.addParam<bool>("output_iteration_info_on_error",
+                        false,
+                        "Set true to output material iteration information when a step fails");
+  params.addParam<Real>(
+      "relative_tolerance", 1e-8, "Relative convergence tolerance for material iteration");
+  params.addParam<Real>(
+      "absolute_tolerance", 1e-20, "Absolute convergence tolerance for material iteration");
```

Comment:
Didn't these options get taken away in a refactor of the return mapping stress model in tensor mechanics?

Reply:
This unfortunately doesn't quite fit into the pattern of SingleVariableReturnMappingSolution, so we can't use it right now. As a follow-on to this, I'd like to either make it so I can use that or use a different procedure that doesn't involve iterations.